lib/foreman/renderer.rb

```diff
@@ -1,389 +1,42 @@
-      :administrator,
-      :proxy_request_timeout,
-      :http_proxy,
-      :http_proxy_except_list,
-      :email_reply_address,
-      :safemode_render,
-      :manage_puppetca,
-      :ignored_interface_identifiers,
-      :remote_addr,
-      :token_duration,
-      :dns_conflict_timeout,
-      :name_generator_type,
-      :default_pxe_item_global,
-      :default_pxe_item_local,
-      :puppet_interval,
-      :outofsync_interval,
-      :default_puppet_environment,
-      :modulepath,
-      :puppetrun,
-      :puppet_server,
-      :update_ip_from_built_request
-    ]
-
-    def template_logger
-      @template_logger ||= Foreman::Logging.logger('templates')
-    end
-
-    def host_enc(*path)
-      check_host
-      @enc ||= @host.info.deep_dup
-      return @enc if path.compact.empty?
-      enc = @enc
-      step = nil
-      path.each { |step| enc = enc.fetch step }
-      enc
-    rescue KeyError
-      raise(HostENCParamUndefined, _('Parameter %{name} is not set in host %{host} ENC output, resolving failed on step %{step}') % { :name => path, :step => step, :host => @host })
-    end
-
-    def host_param(param_name, default = nil)
-      check_host
-      @host.host_param(param_name) || default
-    end
-
-    def host_param!(param_name)
-      check_host_param(param_name)
-      host_param(param_name)
-    end
-
-    def host_puppet_classes
-      check_host
-      @host.puppetclasses
-    end
-
-    def host_param_true?(name)
-      check_host
-      @host.params.has_key?(name) && Foreman::Cast.to_bool(@host.params[name])
-    end
-
-    def host_param_false?(name)
-      check_host
-      @host.params.has_key?(name) && Foreman::Cast.to_bool(@host.params[name]) == false
-    end
-
-    def subnet_has_param?(subnet, param_name)
-      validate_subnet(subnet)
-      subnet.parameters.exists?(name: param_name)
-    end
-
-    def subnet_param(subnet, param_name)
-      validate_subnet(subnet)
-      param = subnet.parameters.where(name: param_name).first
-      param.nil? ? nil : param.value
-    end
+    class << self
+      attr_writer :config
 
-    def render_safe(template, allowed_methods = [], allowed_vars = {}, scope_variables = {})
-      if Setting[:safemode_render]
-        box = Safemode::Box.new self, allowed_methods, template_name
-        erb = ERB.new(template, nil, '-')
-        box.eval(erb.src, allowed_vars.merge(scope_variables))
-      else
-        # we need to keep scope variables and reset them after rendering otherwise they would remain
-        # after snippets are rendered in parent template scope
-        kept_variables = {}
-        scope_variables.each { |k, v| kept_variables[k] = instance_variable_get("@#{k}") }
-
-        allowed_vars.merge(scope_variables).each { |k, v| instance_variable_set "@#{k}", v }
-        erb = ERB.new(template, nil, '-')
-        # erb allows to set location since Ruby 2.2
-        erb.location = template_name, 0 if erb.respond_to?(:location=)
```

Comment:
this has not been moved to unsafe renderer, therefore we're losing the trace in unsafe renderer, could you please put this back and even remove the `if` condition since we already support Ruby 2.3+?

to reproduce, disable safe mode, make a typo in your template, e.g. <% 1 + %> and read the error message, it should contain template name and line number